tests/unit/test_base_path.py

```diff
@@ -0,0 +1,183 @@
+import requests
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+
+auth = UsernamePasswordAuth('username', 'password')
+
+"""
+#This is a test file for testing imperfect base paths, and various other tests too.
+"""
+
+"""
+#The first five tests demonstrate that the client can handle badly written base paths and produce correct final urls.
+#Each test runs a request to the server, and produces the correct final url.
+"""
+
+
+@responses.activate
+def test_base_path_no_trailing_slashes():
+    base_path = "/api/versioned/v1"
+    unify = Client(auth, base_path=base_path)
+    print(unify)
+
+    full_url = "http://localhost:9100/api/versioned/v1/datasets/1"
+    responses.add(responses.GET, full_url, status=200)
+
+    unify.get("datasets/1")
+
+
+@responses.activate
+def test_base_path_no_leading_slashes():
+    base_path = "api/versioned/v1/"
+    unify = Client(auth, base_path=base_path)
+    print(unify)
+
+    full_url = "http://localhost:9100/api/versioned/v1/datasets/1"
+    responses.add(responses.GET, full_url, status=200)
+
+    unify.get("datasets/1")
+
+
+@responses.activate
+def test_base_path_no_slashes():
+    base_path = "api/versioned/v1"
+    unify = Client(auth, base_path=base_path)
+    print(unify)
+
+    full_url = "http://localhost:9100/api/versioned/v1/datasets/1"
+    responses.add(responses.GET, full_url, status=200)
+
+    unify.get("datasets/1")
+
+
+@responses.activate
+def test_base_path_default_slashes():
+    base_path = "/api/versioned/v1/"
+    unify = Client(auth, base_path=base_path)
+    print(unify)
+
+    full_url = "http://localhost:9100/api/versioned/v1/datasets/1"
+    responses.add(responses.GET, full_url, status=200)
+
+    unify.get("datasets/1")
+
+
+@responses.activate
+def test_base_path_no_base_path():
+    base_path = ""
+    unify = Client(auth, base_path=base_path)
```

Comment:
Instead of an empty base path, this should verify proper behavior when `base_path` is not provided, i.e.
```
unify = Client(auth)
```